fenix/docs/Firebase-Cloud-Messaging-for-WebPush.md

```diff
@@ -0,0 +1,45 @@
+# Firebase Cloud Messaging for WebPush
+
+## Testing
+If you want to test WebPush support for features like Send Tab or WebPush for web apps, then you need to follow the instructions below to enable this for your debug application:
+
+1. Download the XML credential file from the [Fenix team's Google Drive](https://drive.google.com/file/d/1DwgmqJTSKOY9vMuW2OmRNDXpIY69thTI/view?usp=drive_link).
+2. Place the file in the directory `fenix/app/src/debug/res/values/`.
+    * This will be compiled into the `fenixDebug` variant.
+3. Run the app and verify you receive log messages in `adb logcat` like the ones below:
+    ```
+    Received a new registration token from push service.
+    Got new Firebase token.
+    ```
+
+## Components and code
+The important components for push to work are the `feature-push` (specifically look at the [README](https://searchfox.org/mozilla-mobile/source/firefox-android/android-components/components/feature/push/README.md)) and `lib-push-firebase`.
+
+For GeckoView, look at [`WebPushEngineIntegration.kt`](https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/push/WebPushEngineIntegration.kt) that connects the `AutoPushFeature` from `feature-push` to the `Engine`.
+
+For Firefox Sync, it's the [`feature-accounts-push`](https://searchfox.org/mozilla-mobile/source/firefox-android/android-components/components/feature/accounts-push).
+
+## Firefox Sync & Send Tab
+If you are testing WebPush support with Firefox Sync, you should also expect to see this log shortly after signing into a Firefox Sync account:
+```
+Created a new subscription: [context data]
+```
+
+## FAQ
+
+**Q. Why are credentials stored in Google Drive?**
+  * Firebase services require a credential file, typically named `google-services.json`, which are retrieved from the Firebase console for all the applications in that project. Firebase uses the `com.google.gms.google-services` Gradle plugin in order to generate an XML file to consume with their SDK for the Android application that matches the app ID in that file (this can have any name).
+    According to [Firebase engineers](https://groups.google.com/forum/#!msg/firebase-talk/bamCgTDajkw/uVEJXjtiBwAJ), it should be safe to commit our `google-services.json` file which contains our configuration of Firebase Cloud Messaging. In fact, the Google I/O app commits [their release version](https://github.com/google/iosched/blob/b428d2be4bb96bd423e47cb709c906ce5d02150f/mobile/google-services.json) of the file to Github as well. 
+    While it is safe to do so, we would not advise it as there are other forks of Fenix that should not mistakenly use these credentials in error. For our case, we use the generated XML file directly in our applications that were created from the `google-services.json`.
+
+**Q. Why can we not add the JSON file (mentioned above) directly in the app?**
```

Comment:
and this could be Q. What are the special requirement which mean we can't directly use google-services.json?